packages/firestore/test/unit/local/index_manager.test.ts

```diff
@@ -37,10 +44,200 @@ describe('IndexedDbIndexManager', () => {
+      schemaVersion: INDEXING_SCHEMA_VERSION
+    });
   });
 
+  async function getIndexManager(
+    user = User.UNAUTHENTICATED
+  ): Promise<TestIndexManager> {
+    const persistence = await persistencePromise;
+    return new TestIndexManager(persistence, persistence.getIndexManager(user));
+  }
+
   genericIndexManagerTests(() => persistencePromise);
+
+  it('can add indexes', async () => {
+    const indexManager = await getIndexManager();
+
+    await indexManager.addFieldIndex(fieldIndex('coll1'));
+
+    const fieldIndexes = await indexManager.getFieldIndexes();
+    expect(fieldIndexes).to.have.length(1);
+    expect(fieldIndexes[0]).to.deep.equal(fieldIndex('coll1', { id: 1 }));
+  });
+
+  it('uses auto-incrementing index id', async () => {
+    const indexManager = await getIndexManager();
+
+    await indexManager.addFieldIndex(fieldIndex('coll1'));
+    await indexManager.addFieldIndex(fieldIndex('coll2'));
+
+    const fieldIndexes = await indexManager.getFieldIndexes();
+    expect(fieldIndexes).to.have.length(2);
+    expect(fieldIndexes[0]).to.deep.equal(fieldIndex('coll1', { id: 1 }));
+    expect(fieldIndexes[1]).to.deep.equal(fieldIndex('coll2', { id: 2 }));
+  });
+
+  it('can get indexes', async () => {
+    const indexManager = await getIndexManager();
+
+    let fieldIndexes = await indexManager.getFieldIndexes('coll1');
+    expect(fieldIndexes).to.have.length(0);
+
+    await indexManager.addFieldIndex(
+      fieldIndex('coll1', { fields: [['value', IndexKind.ASCENDING]] })
+    );
+    await indexManager.addFieldIndex(
+      fieldIndex('coll2', { fields: [['value', IndexKind.CONTAINS]] })
+    );
+
+    fieldIndexes = await indexManager.getFieldIndexes('coll1');
+    expect(fieldIndexes).to.have.length(1);
+    expect(fieldIndexes[0]).to.deep.equal(
+      fieldIndex('coll1', {
+        id: 1,
+        fields: [['value', IndexKind.ASCENDING]]
+      })
+    );
+
+    await indexManager.addFieldIndex(
+      fieldIndex('coll1', { fields: [['newValue', IndexKind.CONTAINS]] })
+    );
+
+    fieldIndexes = await indexManager.getFieldIndexes('coll1');
+    expect(fieldIndexes).to.have.length(2);
+    expect(fieldIndexes[0]).to.deep.equal(
+      fieldIndex('coll1', {
+        id: 1,
+        fields: [['value', IndexKind.ASCENDING]]
+      })
+    );
+    expect(fieldIndexes[1]).to.deep.equal(
+      fieldIndex('coll1', {
+        id: 3,
+        fields: [['newValue', IndexKind.CONTAINS]]
+      })
+    );
+  });
+
+  it('can update collection group', async () => {
+    const indexManager = await getIndexManager();
+
+    await indexManager.addFieldIndex(fieldIndex('coll1'));
+    await indexManager.addFieldIndex(fieldIndex('coll1'));
+    await indexManager.addFieldIndex(fieldIndex('coll2'));
+
+    let fieldIndexes = await indexManager.getFieldIndexes();
+    expect(fieldIndexes[0].indexState).to.deep.equal(
+      new IndexState(0, IndexOffset.min())
```

Comment:
Perhaps explain what 0 and 1 means here.

Reply:
Done